icontrol/session.py

```diff
@@ -86,11 +86,22 @@ def _validate_icruri(base_uri):
     scheme, netloc, path, _, _ = urlsplit(base_uri)
     if scheme != 'https':
         raise InvalidScheme(scheme)
-    if not path.startswith('/mgmt/tm/'):
-        error_message = "The path must start with '/mgmt/tm/'!!  But it's:" +\
+
+    # This can be iWorkflow or BIG-IQ
+    if path.startswith('/mgmt/shared/'):
+        return _validate_prefix_collections(path[13:])
+
+    # This can also be in iWorkflow or BIG-IQ
+    if path.startswith('/mgmt/cm/'):
+        return _validate_prefix_collections(path[9:])
+
+    # Most of the time this is BIG-IP
+    if path.startswith('/mgmt/tm/'):
+        return _validate_prefix_collections(path[9:])
+    else:
+        error_message = "The path must start with '/mgmt/tm/'!  But it's:" +\
```

Comment:
This error message doesn't make sense any more because it can actually start with `/mgmt/shared/`, `/mgmt/cm/`, or `/mgmt/tm/`.

The code also looks a little weird, because you have those if statements and then only one else.  Maybe if else statements would be more readable or setting a variable to the subpath and passing that to the validator at the end or something?